scripts/metadata_comparison/metadata_comparison/lib/operation_ids.py

```diff
@@ -33,30 +33,55 @@ def operation_id_to_api_version(value: str) -> str:
         raise Exception(f'Cannot deduce PAPI api version from unexpected operation ID format \'{value}\'')
 
 
-def find_operation_ids_in_metadata(json_metadata: Mapping[str, Any]) -> Sequence[str]:
-    """Finds all instances of PAPI operations IDs in a workflow"""
-    # Eg given:
-    # {
-    #   "calls": {
-    #     "workflow_name.task_name": [
-    #       {
-    #         "jobId": "projects/broad-dsde-cromwell-dev/operations/01234567891011121314",
-    # ...
-    #
-    # We want to extract "projects/broad-dsde-cromwell-dev/operations/01234567891011121314"
-    papi_operations = []
-
-    def find_operation_ids_in_calls(calls):
-        for callname in calls:
-            attempts = calls[callname]
+# What a JSON Object works out to be.
+JsonObject = Dict[str, Union[Union[None, AnyStr, float], Any]]
+Accumulator = TypeVar('Accumulator')
+OperationId = AnyStr
+CallNameSequence = Sequence[AnyStr]
+
+OperationMappingCallFunction = Callable[[Accumulator, OperationId, CallNameSequence, JsonObject], None]
+
+
+def visit_papi_operations(json_metadata: JsonObject,
+                          call_fn: OperationMappingCallFunction,
+                          initial_accumulator: Accumulator) -> Accumulator:
```

Comment:
This one could definitely do with some long-form python docstring (ala https://www.python.org/dev/peps/pep-0257/)